logstash-core/src/main/java/org/logstash/common/LsQueueUtils.java

```diff
@@ -49,8 +49,7 @@ public static void addAll(final BlockingQueue<JrubyEventExtLibrary.RubyEvent> qu
         final BlockingQueue<JrubyEventExtLibrary.RubyEvent> queue, final int count, final long nanos
     ) throws InterruptedException {
         int left = count;
-        final Collection<JrubyEventExtLibrary.RubyEvent> collection =
-            new HashSet<>(4 * count / 3 + 1);
+        final Collection<JrubyEventExtLibrary.RubyEvent> collection = new ArrayList<>(count);
```

Comment:
out of curiosity, are we sure this is faster/more efficient than using, for example, a LinkedHashSet which would prevent having to check for inclusion in the ArrayList  (which requires iteration) in the `merge` method?

Reply:
oh did not read the description where you actually address this point.